cmd/collector/app/span_processor.go

```diff
@@ -180,23 +182,46 @@ func (sp *spanProcessor) processItemFromQueue(item *queueItem) {
 	sp.metrics.InQueueLatency.Record(time.Since(item.queuedTime))
 }
 
-func (sp *spanProcessor) addCollectorTags(span *model.Span) {
-	if len(sp.collectorTags) == 0 {
-		return
+func (sp *spanProcessor) dedupTags(span *model.Span) {
```

Comment:
just brainstorming how to make this func do as few allocations as possible:
```
init:
   cTagsBits := map[string]uint64 {
      "tag1": 0x1,
      "tag2": 0x2,
      "tag3": 0x4, 
      etc
   }
   
dedupTags(spanTags)
   var skipmask uint64
   for i, tag in spanTags { // efficiency tip: don't actually use `tag` var as it requires copying, use spanTags[i]
      if tag.key in cTagsBits {
         // duplicate found
         if dedupeMode == keepBoth {
             continue
         }
         // replace it in-place based on dedupe-mode requested
         spanTags[i] = ...
         // and label it to be skipped
         skipmask = skipmask | cTagsBits[tag.key]
      }
   }
   for _, key in cTagsBits {
        if mask & cTagsBits[key] == 0 {
            spanTags = append(spanTags, ...) // append to the same slice
        }
   }
```
this limits the number of collector tags to 64, which I think is more than reasonable

Reply:
This is an elegant solution, assuming a 64 tag limit is acceptable.